libopenage/renderer/stages/hud/render_stage.cpp

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023-2023 the openage authors. See copying.md for legal info.
+
+#include "render_stage.h"
+
+#include "renderer/camera/camera.h"
+#include "renderer/opengl/context.h"
+#include "renderer/resources/assets/asset_manager.h"
+#include "renderer/resources/shader_source.h"
+#include "renderer/resources/texture_info.h"
+#include "renderer/shader_program.h"
+#include "renderer/stages/hud/object.h"
+#include "renderer/texture.h"
+#include "renderer/window.h"
+#include "time/clock.h"
+
+
+namespace openage::renderer::hud {
+
+HudRenderStage::HudRenderStage(const std::shared_ptr<Window> &window,
+                               const std::shared_ptr<renderer::Renderer> &renderer,
+                               const std::shared_ptr<renderer::camera::Camera> &camera,
+                               const util::Path &shaderdir,
+                               const std::shared_ptr<renderer::resources::AssetManager> &asset_manager,
+                               const std::shared_ptr<time::Clock> clock) :
+	renderer{renderer},
+	camera{camera},
+	asset_manager{asset_manager},
+	drag_object{nullptr},
+	clock{clock} {
+	renderer::opengl::GlContext::check_error();
```

Comment:
leftover?

Reply:
No, this is done at the start of every render stage initialization (also for world, terrain, etc.).

Reply:
k, but once we have other active renderer backends we need to geberalize^^